ext/RMagick/rmutil.c

```diff
@@ -1268,7 +1367,16 @@ rm_exif_by_entry(Image *image)
             }
             memcpy(str+len, property+5, property_l-5);
             len += property_l - 5;
-            value = GetImageProperty(image,property);
+#if defined(IMAGEMAGICK_7)
+            value = GetImageProperty(image, property, exception);
+            if (rm_should_raise_exception(exception, RetainExceptionRetention))
+            {
+                xfree(str);
+                rm_raise_exception(exception);
```

Comment:
Something I've been wondering about is what happens when the exception is raised. Does this break out of the C code right here? Or does it continue running the function below and then Ruby does some magic to raise the exception in Ruby land?

Reply:
The C code inside `rm_magick_error` calls the raise method of `Class_ImageMagickError`. That raises the exception inside Ruby land and stops the execution of the method.